chore: build auditor in release artifacts workflow

jacderida opened the pull request:
This workflow is useful for testing and I can use it to produce auditor artifacts that can manually be uploaded to S3.

## Description

reviewpad:summary

joshuef commented on .github/workflows/run_pr_checks.yml:
```diff
@@ -85,69 +79,36 @@ jobs:
+          cargo build --release --bin testnet
+      - name: upload artifacts to s3
+        run: |
+          tar -zcvf target-$GITHUB_RUN_ID.tar.gz -C target/ .
```
So we need all these artifacts? I would have just thought we needed `sn_node` and `testnet` bins?

jacderida replied:
Yeah, that's right, and actually that was what I had originally. This was an experiment to see if having the whole target directory could help speed up the slow compilation times for the CLI job. It didn't help though. I didn't change it back because it's lightning fast to copy stuff from an EC2 machine to S3, so I didn't really notice it. I'll change it back now though.

joshuef commented on .github/workflows/run_pr_checks.yml:
```diff
@@ -253,177 +214,115 @@ jobs:
+
+      - name: Build sn bins
+        run: |
+          cd sn_node && cargo build --release --bins
```
Can we not pull this from the prior run?

jacderida replied:
The reason for this was so that this job could start right away rather than wait for the `build` job. It's not really a big deal to build on the EC2 instance, so it meant this job could just go ahead without waiting. I'll change it to be dependent on the build job, but I don't think it'll make a huge difference.

joshuef commented on .github/workflows/run_pr_checks.yml:
```diff
@@ -253,177 +214,115 @@ jobs:
+        run: |
+          cd sn_node && cargo build --release --bins
+
+      - name: Build testnet
```
and this...?

joshuef commented:
I'm not sure EC2 makes sense for CLI (which I think could alllso pull artifacts from a prior build run no?).

At the moment, we're getting a ~18 min speed up for all the tests.... about 40% faster. But from aws billing, we already are paying $50 for that, against... however many test runs we have. So I'm not sure how worthwhile that is (if we had it against all the PRs eg). 

-----

I think @jacderida that if we get only the bins up/down. And use them all over the other jobs... but have those running on GHA infra we're more viable. 

We could then have the split/churn tests go on the AWS infra for the bors `merge` job to further reduce when we need to run those?


----

What do you reckon?

jacderida commented:
> I'm not sure EC2 makes sense for CLI (which I think could alllso pull artifacts from a prior build run no?).
> 
> At the moment, we're getting a ~18 min speed up for all the tests.... about 40% faster. But from aws billing, we already are paying $50 for that, against... however many test runs we have. So I'm not sure how worthwhile that is (if we had it against all the PRs eg).
> 
> I think @jacderida that if we get only the bins up/down. And use them all over the other jobs... but have those running on GHA infra we're more viable.
> 
> We could then have the split/churn tests go on the AWS infra for the bors `merge` job to further reduce when we need to run those?
> 
> What do you reckon?

OK no problem. The CLI job is really slow even with building the `sn_node` and `testnet` binaries on the other machine because it takes so long to compile the `safe` binary and the testing code on the GHA infrastructure. I'll change it back.

jacderida committed:
chore: build auditor in release artifacts workflow

This workflow is useful for testing and I can use it to produce auditor artifacts that can manually
be uploaded to S3.
